### PR TITLE
Merchant edit item #12

### LINKED ISSUE #11: User Story 58, Merchant cannot edit an item if details are bad/missing
```
As a merchant
When I try to edit an existing item
If any of my data is incorrect or missing (except image)
Then I am returned to the form
I see one or more flash messages indicating each error I caused
All fields are re-populated with my previous data
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #19

child of #79

depends on #12

### LINKED ISSUE #12: User Story 57, Merchant edits an item
```
As a merchant
When I visit my items page
And I click the edit button or link next to any item
Then I am taken to a form similar to the 'new item' form
My URI route will be "/dashboard/items/15/edit" (if the item's ID was 15)
The form is re-populated with all of this item's information
I can change any information, but all of the rules for adding a new item still apply:
- name and description cannot be blank
- price cannot be less than $0.00
- inventory must be 0 or greater

When I submit the form
I am taken back to my items page
I see a flash message indicating my item is updated
I see the item's new information on the page, and it maintains its previous enabled/disabled state
If I left the image field blank, I see a placeholder image for the thumbnail
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #19

child of #79

depends on #14

### PR DESCRIPTION
**Pull request by allieIsOkay**
- Adds tests for merchant's ability to edit items
- Updates routing to match user story specifications (/dashboard/items/:id/edit)
- Moves form_for opening tag outside of partial and in to views for new and edit, passing local variable 'f' to partial views for form
- Adds ability for merchant to update items and flash messaging
- Adds error messaging and logic for improper information present in item update form
- closes #12 
- closes #11

### CODE REVIEW

**Review by timnallen — APPROVED** [review]
Nice job Peter!

**Comment by allieIsOkay** [comment]
Updated PR description to close appropriate issues.

File: app/views/merchant/items/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<%= render partial: 'merchant/items/errors' %>
+<%= form_for @item, :url => {:controller => 'merchant/items', :action => 'update'} do |f| %>
+<%= render partial: 'merchant/items/item', locals: {:f => f} %>
```
Comment (teresa-m-knowles) [inline app/views/merchant/items/edit.html.erb:3]: what does the locals: {:f => f} do?

File: spec/features/merchants/items/merchant_edit_item_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      end
+
+      it "I remove the price" do
+        fill_in :Price, with: ""
+        click_button 'Submit'
+        expect(page).to have_content("There are problems with the provided information.")
+        expect(page).to have_content("Price can't be blank")
+      end
+
+      it "I try and enter a negative value for the price" do
```
Comment (teresa-m-knowles) [inline spec/features/merchants/items/merchant_edit_item_spec.rb:57]: good edge case test!

**Review by teresa-m-knowles — APPROVED** [review]
Great tests and use of partials!
